ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +51,417 @@ def test001_test_plan_delete_testunit(self):
+        Archive and restore one record
+        '''
+
+        self.base_selenium.LOGGER.info('Choosing a random testplan table row')
+        row = self.test_plan.get_random_table_row('test_plans:test_plans_table')
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=row)
+        testplan_number = row_data['Test Plan No.']
+        self.test_plan.sleep_small()
+
+        # archive and navigate to archived table
+        self.base_selenium.LOGGER.info('Testplan number: {} will be archived'.format(testplan_number))
+
+        self.base_selenium.LOGGER.info('Selecting the row')
+        self.test_plan.click_check_box(source=row)
+        self.test_plan.sleep_small()
+
+        self.test_plan.archive_selected_items()
+        self.test_plan.get_archived_items()
+
+        archived_row = self.test_plan.search(testplan_number)
+        self.test_plan.sleep_small()
+        self.base_selenium.LOGGER.info('Checking if testplan number: {} is archived correctly'.format(testplan_number))
+        self.assertIsNotNone(archived_row[0])
+        self.base_selenium.LOGGER.info('Testplan number: {} is archived correctly'.format(testplan_number))
+
+        # restore and navigate to active table
+        self.base_selenium.LOGGER.info('Selecting the row')
+        self.test_plan.click_check_box(source=archived_row[0])
+        self.test_plan.sleep_small()
+
+        self.test_plan.restore_selected_items()
+        self.test_plan.get_active_items()
+
+        restored_row = self.test_plan.search(testplan_number)
+        self.base_selenium.LOGGER.info('Checking if testplan number: {} is restored correctly'.format(testplan_number))
+        self.assertIsNotNone(restored_row[0])
+        self.base_selenium.LOGGER.info('Testplan number: {} is restored correctly'.format(testplan_number))
+
+    def test005_archive_restore_test_plan_multiple_records(self):
+        '''
+        LIMS-3506 Case 2
+        Archive and restore multiple records
+        '''
+        self.base_selenium.LOGGER.info('Choosing random multiple testplans table rows')
+        rows = self.test_plan.select_random_multiple_table_rows(element='test_plans:test_plans_table')
+        testplan_rows = rows[0]
+        testplans_numbers = []
+        for row in testplan_rows:
+            testplans_numbers.append(row['Test Plan No.'])
+        self.test_plan.sleep_small()
+
+        # archive and navigate to archived table
+        self.base_selenium.LOGGER.info('Testplan numbers: {} will be archived'.format(testplans_numbers))
+
+        self.test_plan.archive_selected_items()
+        self.test_plan.get_archived_items()
+
+        self.base_selenium.LOGGER.info(
+            'Checking if testplan numbers: {} are archived correctly'.format(testplans_numbers))
+
+        archived_rows = self.test_plan.search_for_multiple_rows(testplans_numbers, 1)
+
+        self.assertIsNotNone(archived_rows)
+        self.assertEqual(len(archived_rows), len(testplans_numbers))
```

Comment:
assert the testplans numbers are exciting here

Reply:
In line 195, there's a call to a method that searches for the testplan numbers that were archived and it returnes row, I think checking the length will be the same, because the length won't be the same, if one of them wasn't archived successfully

Reply:
I'll use filter instead of search